src/App.tsx

```diff
@@ -2,71 +1,94 @@
+import React, { useState, useMemo, useCallback, useRef, useEffect } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
+
+
+function debounce(callback: Function, delay: number = 300) {
+  let timerId: number = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => {
+      callback(...args)
+    }, delay)
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [name, setName] = useState('');
+  const [appliedPerson, setAppliedPerson] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
```

Comment:
The selected person should be passed via props, but currently, it's managed within the component state. Consider refactoring to accept the selected person as a prop.